### PR TITLE
Query Pagination: Update 'showLabel' help text

### LINKED ISSUE #51335: A11y: aria-label text not visible in editor for Pagination Block if "showLabel" is unchecked
### Description

There is a new feature to hide the label of Query Pagination Previous Block and Query Pagination Next Block inside the Query Pagination Block: https://github.com/WordPress/gutenberg/pull/50779

We just tested this feature for accessibility. If I hide the label text, the block's text value will be used as `aria-label` on the previous and next buttons instead. This is fine behaviour.

But from an editor's view, it could be problematic that an editor won't know this unless he checks the frontend for accessibility. The `showLabel` option's label just says: "Toggle off to hide the label text, e.g. 'Next Page'." It does not give any indication that the text value will still be present in the HTML document and accessible for screen reader users. There is also no indication what the visually hidden values are, as they are completely removed from the editor once the option is enabled.

A possible real life scenario:

An editor inserts the Query Pagination Block and plays around with the previous and next labels. They write some funny text or insert Unicode characters because they want to display arrows. Then they notice the option to hide the labels and just use the preset icons instead. This is exactly what they want, and the experimentally changed texts are now hidden. In the frontend everything looks fine for visual users but screen reader users would read the non-appropriate aria-labels, not explaining what the links do.

A suggestion would be to explain to the editor that the values will still be rendered in the HTML document instead of just the current message: "Toggle off to hide the label text, e.g. 'Next Page'.". For example: "Toggle off to visually hide the label text, e.g. 'Next Page'. The text will still be used for screen readers."

We could also even dynamically display the real text values instead of just "e.g. 'Next Page'" in the Block settings.

### Step-by-step reproduction instructions

1. Insert a Query Loop Block.
2. If not present yet, insert a Query Pagination Block.
3. If not present yet, insert a Query Previous Block or Query Next Block.
4. The block's values are "Previous page" and "Next page" by default.
5. Change the values to some other text.
6. Choose an arrow icon in the Block settings of Query Pagination block in the "Arrow" option.
7. Un-check the now visible option "Show label" underneath the arrow option.
8. The changed text values from the Query Previous Block or Query Next Block will not be visible in the editor anymore. To change them, you have to check "Show label" again.

### Screenshots, screen recording, code snippet

# Backend

<div>
grafik
</div>

<div>
grafik
</div>

<div>
grafik
</div>

<div>
grafik
</div>

# Frontend

grafik

### Environment info

- Gutenberg up from [#ede0978](https://github.com/WordPress/gutenberg/commit/ede097824667aa259392dd3c998cf123b1f43738)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #51335.

PR updates Query Pagination block's `'showLabel` attribute helps text. The project eliminates the `toggle` keyword from labels and help texts. See #66369.

I've also removed the example from a string. The results of toggling to options are visible in the Canvas.

## Testing Instructions
1. Open the Site Editor.
2. Select Query Pagination blocks.
3. Switch to the "Arrow" icon style.
4. Confirm that updated help text is displayed.

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 

|Before|After|
|-|-|
|CleanShot 2024-12-18 at 20 21 20|CleanShot 2024-12-18 at 20 23 20|

### CODE REVIEW

File: packages/block-library/src/query-pagination/query-pagination-label-control.js [hunk 1]
```diff
@@ -9,9 +9,7 @@ export function QueryPaginationLabelControl( { value, onChange } ) {
 		<ToggleControl
 			__nextHasNoMarginBottom
 			label={ __( 'Show label text' ) }
-			help={ __(
-				'Toggle off to hide the label text, e.g. "Next Page".'
-			) }
+			help={ __( 'Make label text visible to the users.' ) }
```
Comment (fabiankaegy) [inline packages/block-library/src/query-pagination/query-pagination-label-control.js]: I like that you remove the CTA from this. Makes it align better with all other instances. 

But I have to say I do think the example was useful to know what the label text is actually referring to

**Review by fabiankaegy — APPROVED** [review]
Left one note but nothing blocking :)

File: packages/block-library/src/query-pagination/query-pagination-label-control.js (hunk 1 above)
Comment (Mamaduka) [reply packages/block-library/src/query-pagination/query-pagination-label-control.js]: > I like that you remove the CTA from this. Makes it align better with all other instances.

CTAs doesn't make sense for similar UI controls.

> But I have to say I do think the example was useful to know what the label text is actually referring to

I don't have a strong opinion on the example, happy to append it back. But here's why I removed it:

* This is parent-controlling child blocks, so it's a half example.
* What if you've changed the label to something very different? Would the example still make sense?

**Comment by afercia** [comment]
I'd welcome this improvement, thanks for working on this @Mamaduka 
Worth noting it doesn't actually solves the problem reported on https://github.com/WordPress/gutenberg/issues/51335

The main point on the issue is that the UI doesn't show the labeling of the previous/next any longer when the setting is switched to 'arrows'. As such:
- Users may initially enter dummy or inappropriate labels text for the previous and next links. Screenshot:

Screenshot 2024-12-19 at 09 28 48

- Then switch to the 'arrows' setting. Screenshot:

Screenshot 2024-12-19 at 09 29 12

At this point, users may completely forget they previously entered inappropriate labels text and the UI doesn't expose the labels text in any way.

This PR improves the help text in the UI but I'd like to avoid it closes https://github.com/WordPress/gutenberg/issues/51335 when merged because the reported issue isn't solved yet. Can we change the `Fixes` keyword in the PR description to `See`?.

File: packages/block-library/src/query-pagination/query-pagination-label-control.js (hunk 1 above)
Comment (afercia) [inline packages/block-library/src/query-pagination/query-pagination-label-control.js]: isn't `to the users` implied?

**Review by afercia — APPROVED** [review]
LGTM. Left a small note.

File: packages/block-library/src/query-pagination/query-pagination-label-control.js (hunk 1 above)
Comment (swissspidy) [inline packages/block-library/src/query-pagination/query-pagination-label-control.js]: Now the help text doesn't really provide much help over the label and could even be removed.

Comment (Mamaduka) [reply packages/block-library/src/query-pagination/query-pagination-label-control.js]: Should I change it to - `Make label text visible, e.g. "Next Page"?

Comment (swissspidy) [reply packages/block-library/src/query-pagination/query-pagination-label-control.js]: That would provide more context 👍

**Pull request merged**